packages/generator-adp/src/app/questions/configuration.ts

```diff
@@ -516,8 +516,9 @@ export class ConfigPrompter {
             this.logger.error(validationResult);
             this.appValidationErrorMessage = validationResult;
             this.isApplicationSupported = false;
-            // Continue to the next prompt for extension project.
-            return true;
+            // Continue to the next prompt for extension project only if the project
+            // is NOT cloud project.
+            return this.isCloud ? validationResult : true;
         }
```

Comment:
I think the _easier_ solution would be to check in the `if block` if we are in a cloud scenario. Entering this **extension project flow** `if block` is not correct because you are setting properties that may cause side effects. We should skip this if block as a whole and directly go to the next `if block` where the string will be returned. What do you think?

```suggestion

if (isAppStudio() && isKnownUnsupported && !this.isCloud) {
   this.logger.error(validationResult);
   this.appValidationErrorMessage = validationResult;
   this.isApplicationSupported = false;
   // Continue to the next prompt for extension project
   return true;
}
```

Reply:
Are you sure about this, it changes logic a bit? The `appValidationErrorMessage`, `isApplicationSupported` won't be set, i tested manually and it works - the validation message is correct and the next button is disabled.

Reply:
fixed in e995a497efacb1d1d3d7b3cf9e6ece20144bf926